File: Guestbook.Web/src/app/api/contracts/item-list.ts
export class ItemList<T> {
    /**Page items */
    data: Array<T>;
    /**Requested page nubmer */
    page: number;
    /**Requested page size */
    pageSize: number;
    /**Total number of items */
    total: number;

    constructor(data: any) {
        if (data) {
            for (var i in data) {
                this[i] = data[i];
            }
        }
    }
}

File: Guestbook.WebApi/GuestBook/Contracts/ItemList.cs
using System.Collections.Generic;

namespace GuestBook.WebApi.Contracts
{
    /// <summary>
    /// Represents list of an items.
    /// </summary>
    public class ItemList<TItem>
    {
        /// <summary>
        /// Requested page number
        /// </summary>
        public int Page { get; set; }

        /// <summary>
        /// Requested page size
        /// </summary>
        public int PageSize { get; set; }

        /// <summary>
        /// Total number of items
        /// </summary>
        public int Total { get; set; }

        /// <summary>
        /// Page items
        /// </summary>
        public List<TItem> Data { get; set; }
    }
}
